proc/proc.go

```diff
@@ -46,6 +46,7 @@ type Process struct {
 	packageMap map[string]string
 
 	allGCache               []*G
+	dwarfLock               sync.Mutex
```

Comment:
I think to make this more comprehensive we should instead put this lock on the dwarf reader in `dwarf/reader.go`, and instead of storing the raw `*dwarf.Data` on the `Process` we should just have the reader. That way this lock/unlock logic can all live in the `dwarf/reader` package instead of the caller having to worry about it.